modeshape-schematic/src/main/java/org/infinispan/schematic/internal/CacheSchematicDb.java

```diff
@@ -78,22 +81,15 @@ public boolean containsKey( String key ) {
     }
 
     @Override
-    public SchematicEntry put( String key,
-                               Document document ) {
+    public void put( String key,
+                     Document document ) {
         SchematicEntry newEntry = new SchematicEntryLiteral(key, document);
-        SchematicEntry oldValue = store.put(key, newEntry);
-        return oldValue != null ? removedResult(key, oldValue) : null;
-    }
-
-    protected SchematicEntry removedResult( String key,
-                                            SchematicEntry entry ) {
-        SchematicEntryLiteral literal = (SchematicEntryLiteral)entry;
-        literal.markRemoved(true);
-        return literal;
+        // use storeForWriting because we don't care about the return type - i.e. we're doing a local put
+        storeForWriting.put(key, newEntry);
```

Comment:
This change doesn't maintain the old behavior of marking the old `SchematicEntryLiteral.removed = true`. Is that really not used anymore?